config/routes.rb

```diff
@@ -39,6 +39,7 @@
       resources :totp_registrations, only: %i(new create destroy)
       resources :totp_authentications, only: %i(create)
       resources :promo_registrations, only: %i(index create)
+      resources :payments
```

Comment:
Okay got a lot of food for thought on using resources here, let me dm you on slack.

Reply:
Dan and I discussed and I think we have some ideas on what needs to happen here 👍